internal/iopool/copy.go

```diff
@@ -0,0 +1,26 @@
+package iopool
+
+import (
+	"io"
+	"sync"
+)
+
+type buffer struct {
+	b []byte
+}
+
+var _copyBufSize = 1024 * 32
+var _pool = sync.Pool{
```

Comment:
optional nit: for multiple variables next to each other, consider using var groups

```
var (
  _copyBufSize = ..
  _pool = sync.Pool{
    // ..
  }
)